src/encoder.rs

```diff
@@ -487,6 +488,19 @@ pub struct FrameInvariants<T: Pixel> {
   /// should be ignored. Invalid frames occur when a subgop is prematurely
   /// ended, for example, by a key frame or the end of the video.
   pub invalid: bool,
+  /// Motion vectors to the _original_ reference frames (not reconstructed).
+  /// Used for lookahead purposes.
+  pub mvs: Vec<FrameMotionVectors>,
```

Comment:
I'd make it a separate struct. FrameInvariant will be split sooner or later and having already self-contained struct inside will make our life easier.

Reply:
Which fields do you want to split into a separate struct?

Reply:
All that you are adding I think, you could fit them in a `FrameLookahead` struct maybe?

Reply:
Well, in this case they *are* FrameInvariant, as in once the lookahead is finished they are baked in place, so it's not so bad right now.

Reply:
It would only make sense to split off `mvs` and `lookahead_rec_buffer` as being completely lookahead-specific. However, `mvs` could later, for example, get used as candidates for the real MV search, thus making them not lookahead-specific any more. So idk.

Reply:
I guess you can land now as-is and we can refine later.